### DISCUSSION TITLE
Flattening a PDF

### DISCUSSION THREAD

**Discussion by quantum80** [ImageMagick/ImageMagick#5115] [Help] [answered]
I'm trying to flatten a multipage PDFs that take too long to spool on our construction drawings.  I've been doing this:

magick -density 150 input.pdf -quality 90 output.png

Then I use Adobe to recombine the many PNG files into a PDF.  It does not seem to flatten them, but slightly improves printing efficiency.  Is there a better way?

**Comment by snibgo** [comment]
What do you mean by "flatten"? That normally means compositing a semi-transparent image over some other image, or a constant colour, so the result is fully opaque. Your command doesn't do this.

**Reply by quantum80** [reply]
I'm just wanting to be able to print these drawings faster with a nice top down view and all layers smashed down.  When I flip through the pages some of them hesitate for a while as it loads the images and creates the objects on the screen and spooling times take a very long time.

**Comment by snibgo** [comment]
It seems you have many PDF pages, and you want just one output image. How about:
```
magick -density 150 input.pdf -layers flatten -quality 90 output.png
```

**Reply by quantum80** [reply]
No, i want as many output pages as input.  So 10 pages in, 10 pages out flat.

**Reply by quantum80** [reply]
This is definitely what I want to do, but on each their own individual page.

**Comment by snibgo** [comment]
To flatten each input page against a constant white background:
```
magick -density 150 input.pdf -layers flatten -quality 90 -background White -alpha remove output.png
```
See https://imagemagick.org/script/command-line-options.php#alpha

**Reply by quantum80** [reply]
That works great, except it puts it all in 1 page.  So on my test run I'm using a 10 page PDF, but the output is a single image.  How can I get 10 pages in, and 10 images out?

**Comment by fmw42** [comment, marked as answer]
How about

`magick -density 150 input.pdf -background White -alpha background -alpha off +adjoin -density 90 output.png`

Does that work?

**Reply by quantum80** [reply]
Yes, but should your second -density 90 be -quality 90 instead?

**Reply by Jmuccigr** [reply]
I don't what what your pages look like, but likely using jpeg as output will help reduce file size without losing visible quality.

**Reply by quantum80** [reply]
I'll try that as well, thanks for the tip!  This is saving me a TON of printer time already.  My documents are 11x17 and sometimes 500 pages long and I make 3-4 copies for construction prints.

**Comment by snibgo** [comment]
Sorry, my mistake. I intended:
```
magick -density 150 input.pdf -quality 90 -background White -alpha remove output.png
```

**Reply by quantum80** [reply]
Thanks for the attempts, this one did not work, but the one above did.  I think the +adjoin makes the difference.
